app/controllers/assign.rb

```diff
@@ -0,0 +1,16 @@
+class Assign
+  def self.mentor(mentee, course_id)
+    available_mentors = Mentor.where(available: true)
+    assigned_mentor = available_mentors.find do |mentor|
+      mentor.courses.where(id: course_id)
+    end
```

Comment:
I would suggest adding a scope to mentor for courses.  Mentor.who_teach(courses)
That way you could get to something like
available_mentors.who_teach(course)